scripts/prepare_for_registration_bulk_creation.py

```diff
@@ -0,0 +1,155 @@
+        # parsing_output = {
+        #     'payload_hash': '07eef7ee-fc69-4e3a-9047-8d0e0a4cb909',
+        #     'provider_id': 'osf',
+        #     'schema_id': '60f74311ed3441000141b265',
+        #     'initiator_id': 't47ka',
+        #     'registrations': [
+        #         {
+        #             'external_id': '27093472',
+        #             'csv_raw': '1997,Ford,F250,Black',
+        #             'csv_parsed': {
+        #                 'year': 1997,
+        #                 'make': 'Ford',
+        #                 'model': 'F250',
+        #                 'color': 'Black',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '98709987',
+        #             'csv_raw': '2000,Volkswagen,Golf,White',  # here goes the raw string of one row
+        #             'csv_parsed': {
+        #                 'year': 2000,
+        #                 'make': 'Volkswagen',
+        #                 'model': 'Golf',
+        #                 'color': 'White',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '64510752',
+        #             'csv_raw': '2011,Subaru,WRX,Blue',  # here goes the raw string of
+        #             'csv_parsed': {
+        #                 'year': 2011,
+        #                 'make': 'Subaru',
+        #                 'model': 'WRX',
+        #                 'color': 'Blue',
+        #             },
+        #         },
+        #     ],
+        # }
+
+    logger.info('Preparing OSF database for registration bulk creation ...')
+    initiator_id = parsing_output.get('initiator_id', None)
+    if not initiator_id:
+        logger.error('Missing initiator id')  # internal error
+        return handle_internal_error()
+    initiator = OSFUser.load(initiator_id)
+    if not initiator:
+        logger.error('Initiator not found: [id={}]'.format(initiator_id))  # internal error
+        return handle_internal_error()
+
+    provider_id = parsing_output.get('provider_id', None)
+    if not provider_id:
+        logger.error('Missing provider id')  # internal error
+        return handle_internal_error()
+    provider = RegistrationProvider.objects.filter(_id=provider_id).first()
+    if not provider:
+        logger.error('Provider not found: [id={}] '.format(provider_id))  # internal error
+        return handle_internal_error()
+
+    schema_id = parsing_output.get('schema_id', None)
+    if not schema_id:
+        logger.error('Missing schema id')  # internal error
+        return handle_internal_error()
+    schema = RegistrationSchema.objects.filter(_id=schema_id).first()
+    if not schema:
+        logger.error('Schema not found: [id={}] '.format(provider_id))  # internal error
+        return handle_internal_error()
+
+    upload = RegistrationBulkUpload.create(parsing_output.get('payload_hash', ''), initiator, provider, schema)
```

Comment:
`payload_hash` is something I can provider as a parameter to the task without bothering the parser.

Reply:
I see. This makes more sense. If we decide to separate parsing output from everything else, I believe both initiator and provider should be taken out of the parsing output as well. How about:

```python
prepare_for_registration_bulk_creation(initiator_id, provider_id, payload_hash, parsing_output, dry_run=False)
```

```python
    parsing_output = {
        'schema_id': '60f74311ed3441000141b265',
        'registrations': [
            {
                'csv_raw': '<here goes the raw CSV row as text string>',
                'csv_parsed': csv_parsed,  # a serializable / pickable object 
            },
            ... (more rows) ...
        ],
```

Reply:
Yes, that is what I mean. The parser shouldn't care about `initiator_id` and `provider_id` since it doesn't have those information anyway. That means we can eliminate many of the `if`s in the code. What we can do is to wrap the fetching of the actual models in a `try` block.